src/foam/nanos/controller/ApplicationController.js

```diff
@@ -423,23 +436,21 @@ foam.CLASS({
       // done to start using SectionedDetailViews instead of DetailViews
       this.__subContext__.register(foam.u2.detail.SectionedDetailView, 'foam.u2.DetailView');
 
-      var self = this;
-
       // Reload styling on theme change
       this.onDetach(this.sub('themeChange', this.reloadStyles));
     },
 
     async function initMenu() {
       if ( this.route ) {
-        this.pushMenu(this.route)
+        this.pushMenu(this.route);
       } else  {
         this.pushDefaultMenu();
       }
     },
 
-    function onClientLoad() {
+    async function onClientLoad() {
       let self = this;
-      this.clientPromise.then(async function(client) {
+      await this.clientPromise.then(async function(client) {
```

Comment:
Should not be awaited. onClientLoad() should just be called and any changes due to it should happen in this then()